In [1]:
# dependencies
import pandas as pd

In [2]:
# data url
URL_Framingham_training='https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Training'
URL_Framingham_test='https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Test'

Framingham_test = pd.read_csv('https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Test')
Framingham_training = pd.read_csv('https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Training')

In [5]:
# data preview
Framingham_test.head()

Index  Sex  Age  Educ  Death
0      4    2   52     2      0
1      6    1   48     1      0
2     10    2   46     3      0
3     14    2   49     2      0
4     16    2   63     1      0

In [9]:
Framingham_test.describe()

Index          Sex          Age         Educ        Death
count   2257.000000  2257.000000  2257.000000  2257.000000  2257.000000
mean    5520.255206     1.573770    54.656624     2.011077     0.292424
std     3241.093877     0.494638     9.703423     1.049469     0.454977
min        4.000000     1.000000    33.000000     1.000000     0.000000
25%     2709.000000     1.000000    47.000000     1.000000     0.000000
50%     5440.000000     2.000000    54.000000     2.000000     0.000000
75%     8230.000000     2.000000    62.000000     3.000000     1.000000
max    11328.000000     2.000000    81.000000     4.000000     1.000000

### 1. Use the training set to run a regression predicting Age, based on Sex and Education.  Obtain a summary of the model.  Do both predictors belong in the model?**





In [7]:
import statsmodels.api as sm

X_train = pd.DataFrame(Framingham_training[['Sex', 'Educ']])
X_train = sm.add_constant(X_train)
y_train = pd.DataFrame(Framingham_training[['Age']])

/Users/sean/opt/anaconda3/lib/python3.9/site-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


In [8]:
# OLS model
model01=sm.OLS(y_train, X_train).fit()
model01.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    Age   R-squared:                       0.017
Model:                            OLS   Adj. R-squared:                  0.017
Method:                 Least Squares   F-statistic:                     67.93
Date:                Mon, 31 Jan 2022   Prob (F-statistic):           5.58e-30
Time:                        14:43:49   Log-Likelihood:                -29110.
No. Observations:                7953   AIC:                         5.823e+04
Df Residuals:                    7950   BIC:                         5.825e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         56.3600      0.408    138.025      0.000      55.560      57.160
Sex            0.4309      0.213      2.023      0.043       0.013       0.848
Educ          -1.1845      0.103    -11.446      0.000      -1.387      -0.982
==============================================================================
Omnibus:                      297.738   Durbin-Watson:                   1.426
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              148.843
Skew:                           0.136   Prob(JB):                     4.78e-33
Kurtosis:                       2.387   Cond. No.                         12.1
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

According to the regression model, we can find that the r-squared is pretty small. It represents that 1.7% of the proportion of the variance for the depdent variable that's explained by variables in this linear regression model.

However, the variables Sex and Educ are statistically significant based on both t score and p value.

### 2. Validate the model from the previous exercise

In [10]:
X_test = pd.DataFrame(Framingham_test[['Sex', 'Educ']])
y_test = pd.DataFrame(Framingham_test[['Age']])
X_test = sm.add_constant(X_test)

model01_test=sm.OLS(y_test, X_test).fit()
model01_test.summary()

/Users/sean/opt/anaconda3/lib/python3.9/site-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    Age   R-squared:                       0.020
Model:                            OLS   Adj. R-squared:                  0.019
Method:                 Least Squares   F-statistic:                     22.74
Date:                Mon, 31 Jan 2022   Prob (F-statistic):           1.67e-10
Time:                        14:53:38   Log-Likelihood:                -8308.5
No. Observations:                2257   AIC:                         1.662e+04
Df Residuals:                    2254   BIC:                         1.664e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         55.4202      0.784     70.690      0.000      53.883      56.958
Sex            1.0424      0.409      2.548      0.011       0.240       1.845
Educ          -1.1954      0.193     -6.199      0.000      -1.574      -0.817
==============================================================================
Omnibus:                       97.498   Durbin-Watson:                   1.705
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               51.146
Skew:                           0.190   Prob(JB):                     7.83e-12
Kurtosis:                       2.368   Cond. No.                         12.2
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Based on the summary of the model, R-Squared did not improve that much.

The coefficients for constant, educ are pretty smiliar. The coefficient for the Sex variable increased to 1.0424.

### 3. Use the regression equation to complete this sentence: "The estimated Age equals..."

In [11]:
print("The estimate Age equals ", round(model01.predict(X_test.head(1))[0],2))

The estimate Age equals  54.85


### 4. Interpret the coefficient for Sex

### 5. Interpret the coefficient for Education.

### 6. Find and interpret the value of Standard error.

### 7. Find and interpret r2 (adj)

### 8. Use your formula to make a prediction of predictor values of your choice. Apply 95% prediction interval. Interpret the results